src/UCTNode.cpp

```diff
@@ -347,6 +402,20 @@ class NodeComp : public std::binary_function<UCTNode::node_ptr_t&,
     NodeComp(int color) : m_color(color) {};
     bool operator()(const UCTNode::node_ptr_t& a,
                     const UCTNode::node_ptr_t& b) {
+
+        if (a->get_certain() || b->get_certain()) {
```

Comment:
I just realised this logic is bad - it creates an invalid comparator which doesn't satisfy strict weak ordering.  I'll need to come up with a different strategy to deal with how to order 'certain draw' with respect to other non-certain nodes.